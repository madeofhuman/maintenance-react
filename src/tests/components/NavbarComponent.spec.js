import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import toJson from 'enzyme-to-json';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../../app/components/NavbarComponent';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

Enzyme.configure({ adapter: new Adapter() });

let store;

describe('Navbar component', () => {
  test('renders', () => {
    store = mockStore({
      auth: {
        authenticated: false,
      },
    });
    const wrapper = shallow(<Navbar store={store} />);
    expect(wrapper.exists()).toBe(true);
  });
  test('renders the profile image icon when the user it authenticated', () => {
    store = mockStore({
      auth: {
        authenticated: true,
      },
    });
    const wrapper = mount(
      <BrowserRouter>
        <Navbar store={store} logout={jest.fn()} />
      </BrowserRouter>,
    );
    expect(wrapper.exists()).toBe(true);
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('.profile-icon')).toHaveLength(1);
    wrapper.find('.profile-icon a').simulate('click');
  });
  test('does not render the profile image icon when the user it not authenticated', () => {
    store = mockStore({
      auth: {
        authenticated: false,
      },
    });
    const wrapper = mount(
      <BrowserRouter>
        <Navbar store={store} />
      </BrowserRouter>,
    );
    expect(wrapper.exists()).toBe(true);
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('.profile-icon')).toHaveLength(0);
  });
});
