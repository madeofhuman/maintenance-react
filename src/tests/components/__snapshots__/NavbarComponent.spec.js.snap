// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component does not render the profile image icon when the user it not authenticated 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Connect(Navbar)
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <Navbar
        authenticated={false}
        logout={[Function]}
        store={
          Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
        }
        storeSubscription={
          Subscription {
            "listeners": Object {
              "clear": [Function],
              "get": [Function],
              "notify": [Function],
              "subscribe": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            },
            "unsubscribe": [Function],
          }
        }
      >
        <div
          className="header"
        >
          <span
            className="logo left"
          >
            <Link
              replace={false}
              to="/"
            >
              <a
                href="/"
                onClick={[Function]}
              >
                <Logo>
                  <img
                    alt="logo"
                    src={null}
                  />
                </Logo>
              </a>
            </Link>
          </span>
        </div>
      </Navbar>
    </Connect(Navbar)>
  </Router>
</BrowserRouter>
`;

exports[`Navbar component renders the profile image icon when the user it authenticated 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Connect(Navbar)
      logout={[MockFunction]}
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <Navbar
        authenticated={true}
        logout={[Function]}
        store={
          Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
        }
        storeSubscription={
          Subscription {
            "listeners": Object {
              "clear": [Function],
              "get": [Function],
              "notify": [Function],
              "subscribe": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            },
            "unsubscribe": [Function],
          }
        }
      >
        <div
          className="header"
        >
          <span
            className="logo left"
          >
            <Link
              replace={false}
              to="/"
            >
              <a
                href="/"
                onClick={[Function]}
              >
                <Logo>
                  <img
                    alt="logo"
                    src={null}
                  />
                </Logo>
              </a>
            </Link>
          </span>
          <div
            className="profile-icon right"
          >
            <span
              className="text orange"
            >
              <Link
                onClick={[Function]}
                replace={false}
                to="/"
              >
                <a
                  href="/"
                  onClick={[Function]}
                >
                  Logout
                </a>
              </Link>
            </span>
            <ProfileIcon>
              <img
                alt="profile icon"
                src={null}
              />
            </ProfileIcon>
          </div>
        </div>
      </Navbar>
    </Connect(Navbar)>
  </Router>
</BrowserRouter>
`;
